vllm/model_executor/layers/attention/ops/paged_attn.py

```diff
@@ -121,6 +121,10 @@ def forward_prefix(
         alibi_slopes: Optional[torch.Tensor],
     ) -> torch.Tensor:
         output = torch.empty_like(query)
+        if input_metadata.kv_cache_dtype == "fp8_e5m2":
+            # Convert cache from uint8 to float8_e5m2.
+            key_cache = key_cache.view(dtype=torch.float8_e5m2)
+            value_cache = value_cache.view(dtype=torch.float8_e5m2)
```

Comment:
The key/value cache's data type could be `torch.float8_e5m2` instead of `torch.uint8` at the beginning. So there is no need to convert dtype at this. Could you mind I submit an commit based on your pr, or I could submit a seperate pr?

Reply:
So, could we use torch.float8_e5m2 instead of torch.uint8, including paged attention kernel?

We don't know the reason that #2279 chose torch.uint8 as kvcache tensor dtype. And to avoid any problem after modifying CUDA kernel, we did not do that.



> Could you mind I submit an commit based on your pr, or I could submit a seperate pr?

Sure! I'll add you to my repository as collaborator, and you can commit the modifications into the PR branch chenxu2048:prefix_fp8_kvcache

Reply:
> So, could we use torch.float8_e5m2 instead of torch.uint8, including paged attention kernel?

Yes. We could use `torch.float8_e5m2` in the python files, which will make more clear to users. And we will keep CUDA kernel unchanged.

Reply:
Got it. CUDA kernel casts key_cache and value_cache into fp8e5m2, and does not care about which dtype the tensors are in python files.

Maybe I could modify them by myself.